docs/dev-setup-guide.md

```diff
@@ -1,8 +1,25 @@
-# Develop Logseq
+# Developer Setup Guide
```

Comment:
Looks like renaming files in this PR has broken documentation links. Worth searching and updating broken linnks

Reply:
+1 for this. Better leave the doc structure unchanged. Only apply wording changes.